js/oanda.js

```diff
@@ -0,0 +1,1632 @@
+        };
+        const response = await this.privateGetInstrumentsInstrumentOrderBook (this.extend (request, params));
+        //
+        // {
+        //     orderBook: {
+        //       instrument: 'GBP_USD',
+        //       time: '2022-02-02T15:40:00Z',
+        //       unixTime: '1643816400',
+        //       price: '1.35683',
+        //       bucketWidth: '0.00050',
+        //       buckets: [
+        //          ...
+        //          {
+        //              price: '1.29150',
+        //              longCountPercent: '0.0063',
+        //              shortCountPercent: '0.0000'
+        //          },
+        //          {
+        //              price: '1.29300',
+        //              longCountPercent: '0.0063',
+        //              shortCountPercent: '0.0000'
+        //          },
+        //          {
+        //              price: '1.29350',
+        //              longCountPercent: '0.0000',
+        //              shortCountPercent: '0.0063'
+        //          },
+        //          ..
+        //       ]
+        // }
+        //
+        const orderbookObject = this.safeValue (response, 'orderBook');
+        const timestamp = this.safeTimestamp (orderbookObject, 'unixTime');
+        return this.parseOrderBook (orderbookObject, symbol, timestamp);
+    }
+
+    parseOrderBook (orderbook, symbol, timestamp = undefined, bidsKey = 'Buy', asksKey = 'Sell', priceKey = 'price', amountKey = 0) {
+        const buckets = this.safeValue (orderbook, 'buckets', []);
+        const medianPrice = this.safeString (orderbook, 'price');
+        // const bucketWidth = this.safeValue (orderbook, 'bucketWidth');
+        const bids = [];
+        const asks = [];
+        for (let i = 0; i < buckets.length; i++) {
+            const bucket = buckets[i];
+            const price = this.safeString (bucket, priceKey);
+            const longCountPercent = this.safeString (bucket, 'longCountPercent');
+            const shortCountPercent = this.safeString (bucket, 'shortCountPercent');
+            const volume = Precise.stringAdd (longCountPercent, shortCountPercent);
+            if (Precise.stringGt (medianPrice, price)) {
+                bids.push ([ this.parseNumber (price), this.parseNumber (volume) ]);
+            } else if (Precise.stringLt (medianPrice, price)) {
+                asks.push ([ this.parseNumber (price), this.parseNumber (volume) ]);
+            } else {
+                throw new ExchangeError (this.id + ' parseOrderBook encountered an unrecognized item of bidask: ' + this.json (bucket));
+            }
+        }
+        return {
+            'symbol': symbol,
+            'bids': this.sortBy (bids, 0, true),
+            'asks': this.sortBy (asks, 0),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'nonce': undefined,
+        };
+    }
+
+    async buildOrderRequest (symbol, type, side, amount, price = undefined, params = {}) {
```

Comment:
This method should not be async, please, test the private api in python and in php.

Reply:
iirc I had private api tested in py/php, but let me double check.

Reply:
updates and fixes applied to Oanda, and I confirm now private-tests pass in all of them. Just herokuapp cors fails (ends up with ExchangeNotAvailable). Please review this PR whenever you want, and let me know if there are issues.

(accidentally, while rebasing/resetting this branch, I've removed all commits from this branch, and now it shows as 1 commit).